git/githistory/rewriter_test.go

```diff
@@ -292,6 +293,76 @@ func TestRewriterAllowsAdditionalTreeEntries(t *testing.T) {
 	AssertBlobContents(t, db, tree3, "extra.txt", "extra\n")
 }
 
+func TestHistoryRewriterCallbacks(t *testing.T) {
+	type Call struct {
+		Type string
+		Path string
+	}
+
+	var calls []*Call
+
+	db := DatabaseFromFixture(t, "linear-history.git")
```

Comment:
It might be good to have a nested subtree in here somewhere

Reply:
Thanks for the suggestion, I caught that in https://github.com/git-lfs/git-lfs/pull/3114/commits/3c574d0ba1e7530866993125918694a1864036e8 and https://github.com/git-lfs/git-lfs/pull/3114/commits/22801fdaeb64c8352b9834d88dd1a65377ad1874.